src/Admin/AbstractAdmin.php

```diff
@@ -1122,33 +1127,52 @@ public function generateMenuUrl($name, array $parameters = [], $absolute = Routi
         return $this->routeGenerator->generateMenuUrl($this, $name, $parameters, $absolute);
     }
 
+    public function setTemplateRegistry(TemplateRegistryInterface $templateRegistry)
+    {
+        $this->templateRegistry = $templateRegistry;
+    }
+
+    /**
+     * @deprecated Use TemplateRegistry services instead
+     *
+     * @param array $templates
+     */
     public function setTemplates(array $templates)
     {
-        $this->templates = $templates;
```

Comment:
I think this could/should be kept for BC (if an admin accesses that property, it will still be filled properly). WDYT @sonata-project/contributors ?